### DISCUSSION TITLE
Could the found optimizations be integrated to pnpm?

### DISCUSSION THREAD

**Discussion by zkochan** [nachoaldamav/snpm#102] [General]
From the description, it looks like ultra works similarly to how pnpm woks. It creates a node_modules without symlinks but pnpm has also an option to create a flat node_modules (via using `node-linker=hoisted`). So I am curious what is the motivation to work on a new package manager instead of contributing to pnpm. And whether the found optimizations would work in pnpm as well.

Now I realized this should've probably been a discussion not an issue. Feel free to convert it.

**Comment by nachoaldamav** [comment]
Hello, first of all I want to clarify that Ultra is based a lot on PNPM, as you can see in how it works, when I was working on Ultra I used a lot the PNPM documentation to learn a lot of things.

About the question of why create a new package manager and not contribute to PNPM, is because the development of this was something casual, I didn't plan to release this until I started with the benchmarks.
As you can see in the [README history](https://github.com/nachoaldamav/ultra/blob/81a1bd32cd8eb5dfe7dc1a7b307582dead8936ef/README.md?plain=1#L17) I started this project to learn about package managers.

In the next few days I will try to take a closer look at the PNPM code and try to implement some of the improvements I have implemented in this project.

I'm going to turn this issue into a discussion in case anyone wants to add something.

**Reply by zkochan** [reply]
I see, feel free to connect with me on discord or PM me on twitter if you want to discuss the details. Or if you have questions about pnpm.
